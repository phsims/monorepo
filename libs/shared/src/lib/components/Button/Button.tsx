import { ReactNode } from 'react';

/* eslint-disable-next-line */
export interface ButtonProps {
  id?: string;
  text: string;
  type?: 'button' | 'submit' | 'reset';
  btnClass?: string;
  disabled?: boolean;
  children?: ReactNode;
}

export function Button({
  text,
  type = 'button',
  disabled = false,
  btnClass = 'group relative flex w-full justify-center rounded-md border border-transparent bg-primary py-2 px-4 text-sm font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2',
  children,
  id = `${type}-button`,
}: ButtonProps) {
  return (
    <button type={type} className={btnClass} id={id} disabled={disabled}>
      {children}
      {text}
    </button>
  );
}

export default Button;
