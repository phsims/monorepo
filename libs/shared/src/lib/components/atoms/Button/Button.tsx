import { ReactNode } from 'react';

/* eslint-disable-next-line */
export interface ButtonProps {
  id?: string;
  text: string;
  type?: 'button' | 'submit' | 'reset';
  btnClass?: string;
  disabled?: boolean;
  children?: ReactNode;
}

export function Button({
  text,
  type = 'button',
  disabled = false,
  btnClass = 'text-xl font-medium bg-primarybg text-primary hover:text-white hover:bg-primary disabled:bg-slate-200 disabled:text-grey disabled:border-slate-500',
  children,
  id = `${type}-button`,
}: ButtonProps) {
  return (
    <button
      type={type}
      className={`rounded-full p-4 lg:px-8  ${btnClass}`}
      id={id}
      disabled={disabled}
    >
      {children}
      {text}
    </button>
  );
}

export default Button;
