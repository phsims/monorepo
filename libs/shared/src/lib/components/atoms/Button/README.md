<!-- README.md -->



Primary UI component for user interaction

```js
import { Button } from "@shared";
```