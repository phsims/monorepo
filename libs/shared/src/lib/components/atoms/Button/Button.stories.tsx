import type { Meta, StoryObj } from '@storybook/react';
/* eslint-disable-next-line */
import { Markdown } from '@storybook/blocks';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import { Button } from './Button';
import ReadMe from './README.md?raw';

const meta: Meta<typeof Button> = {
  component: Button,
  title: '@shared/atoms/Button',
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: {
        component: `<Markdown>${ReadMe}</Markdown>`,
      },
    },
  },
  decorators: [
    (Story) => (
      <div className="max-w-md">
        <Story />
      </div>
    ),
  ],
};
export default meta;
type Story = StoryObj<typeof Button>;

export const Default: Story = {
  args: { text: 'button text' },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/button text/gi)).toBeTruthy();
  },
};

export const Disabled: Story = {
  args: {
    text: 'button text',
    disabled: true,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/button text/gi)).toBeDisabled();
  },
};
