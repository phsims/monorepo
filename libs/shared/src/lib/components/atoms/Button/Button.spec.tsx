import { render } from '@testing-library/react';

import Button from './Button';
const props = {
  text: 'button text',
};
describe('Button', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Button {...props} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<Button {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
