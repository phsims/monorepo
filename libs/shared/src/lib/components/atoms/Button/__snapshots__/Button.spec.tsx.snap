// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<body>
  <div>
    <button
      class="rounded-full p-4 lg:px-8  text-xl font-medium bg-primarybg text-primary hover:text-white hover:bg-primary disabled:bg-slate-200 disabled:text-grey disabled:border-slate-500"
      id="button-button"
      type="button"
    >
      button text
    </button>
  </div>
</body>
`;
